pkg/k8sinit/schema.go

```diff
@@ -155,6 +155,9 @@ type Configuration struct {
 
 	// Join configuration. Setting this will attempt to join the local node to an already existing MicroK8s cluster.
 	Join JoinConfiguration `yaml:"join"`
+
+	// NetworkConfig is configuration of network such us IPv4/v6 cluster and service CIDRs.
+	NetworkConfig map[string]*string `yaml:"networkConfig"`
```

Comment:
If we are in fact going to be calling this `networkConfig`, I'd much rather we instead use a strongly-typed struct to provide this configuration, e.g.
```go
type NetworkConfiguration struct {
	CNI             string // `calico`, `none`
	EnableIPv4      bool
	IPV4ClusterCIDR string
	IPV4ServiceCIDR string
	EnableIPv6      bool
	IPV6ClusterCIDR string
	IPV6ServiceCIDR string
}
```

That way, users that mistype a config get a much better error message.


A much less involved alternative would be to simply call this `extraCNIEnv`, and then document the supported options elsewhere.

Reply:
Lets go with the extraCNIEnv option for now because I am not sure how many/which field we will eventually need to configure the CNI.